File: helixtrajectorycpp/src/cpp/Drivetrain.cpp
#include "Drivetrain.h"

#include <casadi/casadi.hpp>

namespace helixtrajectory {

    Drivetrain::Drivetrain(double mass, double momentOfInertia, const Obstacle& bumpers)
        : mass(mass), momentOfInertia(momentOfInertia), bumpers(bumpers) {
    }

    Drivetrain::~Drivetrain() {
    }
}

File: helixtrajectoryj/src/main/java/org/team2363/helixtrajectory/Drivetrain.java
// Copyright (c) TrajoptLib contributors

package org.team2363.helixtrajectory;

import java.util.Objects;

public abstract class Drivetrain {
    public final double mass;
    public final double momentOfInertia;
    public final Obstacle bumpers;

    protected Drivetrain(double mass, double momentOfInertia, Obstacle bumpers) throws NullPointerException {
        this.mass = mass;
        this.momentOfInertia = momentOfInertia;
        this.bumpers = Objects.requireNonNull(bumpers, "Drivetrain bumpers cannot be null");
    }
}
